mobile/src/calendar/calendar.tsx

```diff
@@ -19,7 +20,16 @@ interface CalendarDispatchProps {
     prevCalendarPage: () => void;
 }
 
-export class CalendarImpl extends Component<CalendarProps & CalendarDispatchProps> {
+export class CalendarImpl extends Component<NavigationScreenProps & CalendarProps & CalendarDispatchProps> {
+
+    public componentDidMount() {
+        if (this.props.navigation) {
+            this.props.navigation.setParams({
```

Comment:
can't you dispatch something instead?

Reply:
Not sure what exactly do you mean. Dispatching events like 'ScrollToTop' from Tabbar?

Reply:
I don't like that we need to pass navigation property everywhere. react-navigation provides some redux actions for common operations, like navigateTo, or go back. Is it possible to set these params using that method? Also, what happens when this parameter is set once and then we go away from the screen. Is it reset? Because the navigation instance is same.

Reply:
It seems this parameters cannot be set using redux navigation actions as the tab bar press callback contains only NavigationRoute. 

As the navigation instance is the same, these params will be the same unless somebody manually rewrite them.

Reply:
in that case, does it make sense to set these parameters once in the root component?

Reply:
or they differ

Reply:
seems like they differ, ok, i understand.